signup.php

```diff
@@ -4,40 +4,60 @@
 
 include('config/init.php');
 echoHeader('Sign up');
-if(isset($_REQUEST['newSignUp'])){
-    insertUser(
-        $_REQUEST['username'], $_REQUEST['password'], $_REQUEST['name'], $_REQUEST['email']
+$errors=array();
+global $errors;
 
-    );
+if(isset($_REQUEST['newSignUp'])){
+    
+$username = $_POST['username'];
+$email = $_POST['email'];
+$password = $_POST['password'];
+$name = $_POST['name'];
+   
+    if(empty($username)){
+        array_push($errors, 'Username is required');
+    }
+    if(empty($password)){
+        array_push($errors, 'Password is required');
+    }
+    if(empty($name)){
+        array_push($errors,'Name is required');
+    }
+    /*if(empty($email)){
+        array_push($errors,'Email is required');*/
+    
+if(count($errors) == 0){
+        insertUser(
+            $_REQUEST['username'], sha1($_REQUEST['password']), $_REQUEST['name'], $_REQUEST['email']);
+        header('location:signup.php');
```

Comment:
It's best to put an `exit` after redirects like this. It normally doesn't matter, but without the exit, in theory code can keep running lower on this page even though a redirect is happening.